proposals/text/0000-evaluation-haystack-2.md

```diff
@@ -0,0 +1,223 @@
+- Title: Evaluation in Haystack 2.0
+- Decision driver: (Silvano Cerza, Julian Risch)
+- Start Date: 2023-08-23
+- Proposal PR: #5794
+- Github Issue or Discussion: https://github.com/deepset-ai/haystack/issues/5265
+
+# Summary
+
+Given the below requirements we redefine how evaluation and metrics are handled in Haystack 2.x.
+
+Our goal is to lower the barrier of entry for new comers but also making it more flexible and extensible for more advanced and expert users.
+All this while making it more modular and easier to test and maintain.
+
+The requirements are:
+
+- compare the performance of different pipelines on level of pipeline outputs (user perspective, integrated eval)
+  - while running the full pipeline we can store intermediate results and calculate metrics for each component that returns answers or documents
```

Comment:
Is it feasible(and makes sense?) to support pipeline replicas for large-scale evaluations? One potential strategy could be to have the evaluation process stateless, thereby preventing the storage of interim results within a pipeline and ensuring all results from components are propagated through the pipeline.